### PR TITLE
feat: send DISCARD command when conn recycled

### PR DESCRIPTION
**Pull request by SoulPancake**
Fixes - #579

### CODE REVIEW

**Comment by SoulPancake** [comment]
@rueian Can you please take a look?

File: internal/cmds/cmds.go [hunk 1]
```diff
@@ -76,6 +76,12 @@ var (
 		cs: newCommandSlice([]string{"UNSUBSCRIBE", "+sentinel", "+slave", "-sdown", "+sdown", "+switch-master", "+reboot"}),
 		cf: noRetTag,
 	}
+
+	// DiscardCmd is predefined DISCARD
+	DiscardCmd = Completed{
+		cs: newCommandSlice([]string{"DISCARD"}),
+		cf: noRetTag,
```
Comment (rueian) [inline internal/cmds/cmds.go:83]:
```suggestion
```

We can’t set the noRetTag on it. The tag will trigger a special handling for PUBSUB commands.

**Review by rueian — APPROVED** [review]

File: pipe_test.go [hunk 2]
```diff
@@ -4129,6 +4129,11 @@ func TestPipe_CleanSubscriptions_6(t *testing.T) {
 			{typ: '+', string: "punsubscribe"},
 			{typ: '_'},
 			{typ: ':', integer: 2},
+		}},
+		RedisMessage{typ: '>', values: []RedisMessage{
```
Comment (rueian) [inline pipe_test.go:4133]: @SoulPancake, I found this reply is incorrect. DISCARD should reply a OK message instead of a push message.

**Review by rueian — CHANGES_REQUESTED** [review]

File: pipe_test.go (hunk 2 above)
Comment (SoulPancake) [reply pipe_test.go]: My bad, Addressed @rueian

**Review by rueian — APPROVED** [review]

**Pull request merged**